homeassistant/components/zabbix/__init__.py

```diff
@@ -159,7 +163,25 @@ def event_to_metrics(
             )
             metrics.append(metric)
 
-        string_keys.update(strings)
+        if include_strings:
+            string_keys_count = len(string_keys)
+            string_keys.update(strings)
+            if len(string_keys) != string_keys_count:
+                strings_discovery = [
+                    {"{#KEY}": string_key} for string_key in string_keys
+                ]
+                metric = ItemValue(
+                    publish_states_host,
+                    "homeassistant.strings_discovery",
+                    json.dumps(strings_discovery),
+                )
+                metrics.append(metric)
+            for key, value in strings.items():
+                metric = ItemValue(
+                    publish_states_host, f"homeassistant.string[{key}]", value
+                )
+                metrics.append(metric)
```

Comment:
I wonder if we instead of duplicating the code above here we could move it to a method `generate_metrics(keys, attributes, type)` and use it for both?

Reply:
Potentially - I wanted to keep the existing code as is to not break anything. Also the implementation of this PR is very well tested by me and I would not like to risk the change...

Reply:
I agree with @abmantis, this level of duplication is not OK
![image](https://github.com/user-attachments/assets/c96e577b-52ca-4091-9234-97850997bd0e)

> the implementation of this PR is very well tested by me and I would not like to risk the change
This is not a valid reason for not deduplicating the code. The code might be well tested right now, but with duplication there's a very significant risk of changes in the future where the treatment of string and numerical states accidentally diverges.